pkg/cloudproviders/gcp/auth/client_manager_impl.go

```diff
@@ -40,7 +43,7 @@ func NewSTSClientManager(namespace string, secretName string) STSClientManager {
 	}
 	mgr := &stsClientManagerImpl{storageClientHandler: storage.NewClientHandlerNoInit()}
 	mgr.credManager = newCredentialsManagerImpl(k8sClient, namespace, secretName, mgr.updateClients)
-	mgr.updateClients()
+	go mgr.updateClients()
```

Comment:
Do we want to do this asynchronously and risk having clients that aren't updated yet?
It doesn't seem to me like the initial credentials fetching is expensive, furthermore it might indicate wrong results for e.g. the test functionality in case a non-updated client is used.

Reply:
My thinking is that the initial credentials fetching actually is expensive (well kind of). When running on GCP, `google.FindDefaultCredentials` reaches out to the meta data service, which apparently was enough to fail the 10s timeout in the UI e2e tests. Since this is run on Central startup, I think it's unlikely that something clicks the `Test` button. But I'm fine with leaving it in the Central main loop, the problem with the UI tests was that it was run despite on the first API call.